sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1069,15 +1126,8 @@ private void releaseChannel(final Channel channel, final Promise<Void> promise)
                 }
             }
         } else {
-            final IllegalStateException error = new IllegalStateException(lenientFormat(
```

Comment:
There are two possible causes either mis-managed pool or non-acquired connection. 

For at-least mis-managed pool, channel.Close() will ensure at-least the respective pool also will release the place holders.
For non-acquired issue tracing LGTM (please adjust the test little)